src/mpi/coll/bcast/bcast_init.c

```diff
@@ -11,66 +11,24 @@
 #include "../ibcast/ibcast_tsp_tree_algos_prototypes.h"
 #include "../ibcast/ibcast_tsp_scatterv_allgatherv_algos_prototypes.h"
 
-static int MPIR_Bcast_sched_intra_auto(void *buffer, MPI_Aint count, MPI_Datatype datatype,
-                                       int root, MPIR_Comm * comm_ptr, MPIR_TSP_sched_t * sched)
-{
-    int mpi_errno = MPI_SUCCESS;
-
-    mpi_errno = MPIR_TSP_Ibcast_sched_intra_tree(buffer, count, datatype, root, comm_ptr,
-                                                 MPIR_Ibcast_tree_type, MPIR_CVAR_IBCAST_TREE_KVAL,
-                                                 MPIR_CVAR_IBCAST_TREE_PIPELINE_CHUNK_SIZE, sched);
-
-    return mpi_errno;
-}
-
 int MPIR_Bcast_init_impl(void *buffer, MPI_Aint count, MPI_Datatype datatype, int root,
                          MPIR_Comm * comm_ptr, MPIR_Info * info_ptr, MPIR_Request ** request)
 {
     int mpi_errno = MPI_SUCCESS;
-    MPIR_TSP_sched_t *sched;
 
     /* create a new request */
     MPIR_Request *req = MPIR_Request_create(MPIR_REQUEST_KIND__PREQUEST_COLL);
     MPIR_ERR_CHKANDJUMP(!req, mpi_errno, MPI_ERR_OTHER, "**nomem");
 
+    MPIR_Comm_add_ref(comm_ptr);
```

Comment:
I would move `bcast_init.c` in a separate commit.  The rest of this commit is pretty much refactoring `ibcast`.

Reply:
done.